src/main/java/apoc/number/Number.java

```diff
@@ -22,76 +22,97 @@
-  public Stream<StringResult> formatDouble(final @Name("number") double number, @Name("pattern") String pattern, @Name("language") String language) {
-    DecimalFormat format = buildFormatter(pattern, language);
-    return Stream.of(new StringResult(format.format(number)));
+  @Procedure("apoc.number.format.pattern")
+  @Description("apoc.number.format.pattern(number, pattern) yield value | format a long or double using a pattern and the default system language to produce a string")
+  public Stream<StringResult> formatByPattern(final @Name("number") Object number, final @Name("pattern") String pattern) {
+    return formatByPatternAndLanguage(number, pattern, null);
   }
   
-  @Procedure
-  @Description("apoc.number.parseAsLong(text, pattern, locale) yield value | parse a text using a (optional) pattern and for a (optional) locale to produce a long")
-  public Stream<LongResult> parseAsLong(final @Name("text") String text, @Name("pattern") String pattern, @Name("language") String language) throws ParseException {
-    DecimalFormat format = buildFormatter(pattern, language);
-    return Stream.of(new LongResult(format.parse(text).longValue()));
+  @Procedure("apoc.number.format.lang")
+  @Description("apoc.number.format.lang(number, lang) yield value | format a long or double using the default system pattern pattern and a language to produce a string")
+  public Stream<StringResult> formatByLanguage(final @Name("number") Object number, final @Name("lang") String lang) {
+    return formatByPatternAndLanguage(number, null, lang);
   }
-
-  @Procedure
-  @Description("apoc.number.parseAsDouble(text, pattern, locale) yield value | parse a text using a (optional) pattern and for a (optional) locale to produce a double")
-  public Stream<DoubleResult> parseAsDouble(final @Name("text") String text, @Name("pattern") String pattern, @Name("language") String language) throws ParseException {
-    DecimalFormat format = buildFormatter(pattern, language);
-     return Stream.of(new DoubleResult(format.parse(text).doubleValue()));
+  
+  @Procedure("apoc.number.format.pattern.lang")
+  @Description("apoc.number.format.pattern.lang(number, pattern, lang) yield value | format a long or double using a pattern and a language to produce a string")
+  public Stream<StringResult> formatByPatternAndLanguage(final @Name("number") Object number, final @Name("pattern") String pattern, final @Name("lang") String lang) {
+    validateNumberParam(number);
+    DecimalFormat format = buildFormatter(pattern, lang);
+    return Stream.of(new StringResult(format.format(number)));
+  }
+  
+  // Parse As Long
+  
+  @Procedure("apoc.number.parseAsLong")
+  @Description("apoc.number.parseAsLong(text) yield value | parse a text using the default system pattern and language to produce a long")
+  public Stream<LongResult> parseAsLong(final @Name("text") String text) throws ParseException {
```

Comment:
`ResultObject` implies that the caller will have to manage the "cast" to `long` or `double`